src/Idris/Parser.idr

```diff
@@ -1501,28 +1502,27 @@ import_ fname indents
 export
 prog : FileName -> SourceEmptyRule Module
 prog fname
-    = do b <- bounds (do doc    <- option "" documentation
-                         nspace <- option (nsAsModuleIdent mainNS)
-                                     (do keyword "module"
-                                         moduleIdent)
-                         imports <- block (import_ fname)
-                         pure (doc, nspace, imports))
-         ds      <- block (topDecl fname)
-         (doc, nspace, imports) <- pure b.val
-         pure (MkModule (boundToFC fname b)
+    = do p1 <- position
```

Comment:
Why the switch to `position` inplace of `bounds`? The former returns the position of the next token, and probably it is the cause of the relaxed bounds in some test cases, as mentioned also by @gallais

Reply:
This was to avoid computing the bounds on all the following parser which are not guaranteed to consume. I tried removing the position and computing the bounds anyways (by returning `b ** WithBounds b tok` on `option`) but I still get block with bounds outside of the file in `"interface016"`